nav2_behavior_tree/plugins/action/spin_action.cpp

```diff
@@ -36,12 +36,22 @@ class SpinAction : public BtActionNode<nav2_msgs::action::Spin>
     const std::string & action_name,
     const BT::NodeConfiguration & conf)
   : BtActionNode<nav2_msgs::action::Spin>(xml_tag_name, action_name, conf)
+  {
+  }
+
+  void on_tick() override
```

Comment:
Since these are static parameters in the xml, do we need to run these in `on_tick`?

Reply:
We would need them there if we want to send a new goal as preemption (the null goal I made), to reset the goal whenever we enter the node.
However, if I change this to simply cancel the recovery, it can stay in the constructor.